### ISSUE TITLE
Implement default "never" option for `check-line-alignment` rule: linting inside comments block spaces/tab to single space

### ISSUE THREAD

**Issue by jperocho** [gajus/eslint-plugin-jsdoc#483]
Just a question if this plugin has a rule to check for multiple spaces or tabs inside a comment block?

```
 /**
 * Some jsdocs for a function
 * @param   {String}   argument string - string
 * @param   {Object}   property        - property object
 * @returns {Function} callback        - callback
 */
```

Thanks in advance.

**Comment by brettz9** [comment]
Not currently. One can use regexes on the description portion using `match-description`, but that wouldn't catch whitespace before or after the tag, type, or name (or default value).

**Comment by l1bbcsg** [comment]
This is quite complex given the lack of concrete syntax tree or parsers that can produce one for JSDoc.
Regular expressions will quickly turn into a mess if edge cases and more scenarios were to be implemented.

Neither Doctrine, nor comment-parser can do what espee does for js, namely tokenize input and keep information about all whitespaces and non-syntax important stuff.
When we previously discussed this in my company, we figured we're going to need to write a new JSDoc parser that either keeps track of tokens or builds a concrete syntax tree, but it wasn't the task we had time for.
This would allow all kinds of validation of JSDoc codestyle – indentation, whitespace placement, you name it.

Some attempt were made in [eslint-plugin-interfaced](https://github.com/interfaced/eslint-plugin-interfaced) however, particularly [JSDocTokenizer](https://github.com/interfaced/eslint-plugin-interfaced/blob/master/lib/jsdoc-tokenizer.js) and some [helper functions](https://github.com/interfaced/eslint-plugin-interfaced/blob/master/lib/ast-utils.js). And [`jsdoc-type-spacing`](https://github.com/interfaced/eslint-plugin-interfaced/blob/master/docs/rules/jsdoc-type-spacing.md) is an example of a rule that validates whitespaces.

**Comment by yvele** [comment]
What about the use case where we want to **disable multi space alignment** in JSDoc? Does that feature requires complex parsing?

Something that fixes:

```js
/**
 * @param   {string}   foo   This is foo.
 * @param   {object}   bar   This is bar.
 * @returns {boolean}        This is the returned value.
 */
```

to:

```js
/**
 * @param {string} foo This is foo.
 * @param {object} bar This is bar.
 * @returns {boolean} This is the returned value.
 */
```

The rule may be very specific like

```json
"jsdoc/no-aligment" : "error"
```

Or a more configurable rule similar to https://eslint.org/docs/rules/key-spacing but it looks lke we are limited by the JSDoc parser.

So.. can we at least have a simple rule to disallow multi space alignment?


PS: As a workaround can we have a generic `match` regex that applies to a whole JSDoc line? 🤔 with a custom message and fix of course.

**Comment by brettz9** [comment]
@yvele : I think based on the title of the issue that that's what the original poster intended for the rule, though there could conceivably be a rule which automatically added whitespace with alignment. I'd be personally open to a whole-line regex solution as an interim solution if someone could put a PR together, though it'd be great as @l1bbcsg suggests if we could get AST for jsdoc, or at least build on `comment-parser` further to allow full round-tripping.

**Comment by brettz9** [comment]
I've renamed this issue to reflect the fact that we've made space for this functionality within a new rule `check-line-alignment`, but currently the implementation is only able to do the opposite. The "always" option will do the opposite, but the functionality for this issue should provide the "never" option and is even reserved already as the default behavior. At such time as it may be implemented, we can also add this rule to the "recommended" rules.

**Comment by brettz9** [comment]
I've gone ahead and implemented the "never" behavior as @renatho previously implemented "always" behavior.

It'd probably be better if we switched internally for the "always" behavior to the new `comment-parser` tokens in place of our custom regexes, and we don't have any specified behavior for alignment of multi-line descriptions, but the main behavior is now available in v31.0.2. Thanks for the report and to @renatho for getting the feature going!
